src/main/java/robot/subsystems/drivetrain/pure_pursuit/PurePursue.java

```diff
@@ -8,20 +8,34 @@
  *
  */
 public class PurePursue extends Command {
-    private double lastLeftEncoder;
-    private double lastRightEncoder;
-    private double initAngle;
-    private Path path;
-    private Point currentPoint;
-    private Drivetrain drive;
-
+    private Drivetrain drive; //Instance of the subsystem
+    private Path path; //Command specific path to follow
+    private Point currentPoint; //holds X and Y variables for the robot
+    private Point currentLookahead; //holds X and Y variables for the Lookahead point
+    private int direction; //whether the robot drives forward or backwards (-1 or 1)
+    private double lastLeftSpeed; //the last speed of the left encoder
+    private double lastRightSpeed; //the last speed of the right encoder
+    private double lastLeftEncoder; //the last distance of the left encoder
+    private double lastRightEncoder; //the last distance of the right encoder
+    //private double initAngle;
+    private double lastLookaheadDistance; //distance of the last lookahead from the start of the path
+    private double Kp, Ka, Kv;
+    private double lookaheadRadius;
     /**
-     * @param path
+     * A command class.
+     *
+     * @param path       the Path class that the robot is going to follow
+     * @param isReversed states if the robot should drive forward or backwards along the path.
      */
-    public PurePursue(Path path) {
+    public PurePursue(Path path, boolean isReversed, double lookaheadRadius, double Kp, double Ka, double Kv) {
+        this.lookaheadRadius = lookaheadRadius;
+        this.Kp = Kp;
+        this.Ka = Ka;
+        this.Kv = Kv;
+        direction = isReversed ? -1 : 1;
         drive = Robot.drivetrain;
         this.path = path;
-        currentPoint = new Point(0,0);
+        currentPoint = new Waypoint(drive.robotPoint.getX(), drive.robotPoint.getY());
         // Use requires() here to declare subsystem dependencies
         // eg. requires(chassis);
```

Comment:
This command is lonely 😭! Maybe give it a friend, like a subsystem 😁?
You forgot the `requires()` method.
```suggestion
       requires(Robot.drivetrain);
```
Also, make sure to actually add the `drivetrain` object to the [`Robot`.java](https://github.com/DanKatzuv/BaseBot/blob/pure-pursue/src/main/java/robot/Robot.java) file.